### PR TITLE
Gradle-ized

### PR DESCRIPTION
**Pull request by AgoloKennethTan**
Gradle version: `0.1.2`

What I have done:
- Reverted the code to reflect what was done in the source code as much as possible. Also, using the configuration that is used in Pyramid.
- Added comments on these reverts.
- Wrote a simple smoke test to make sure things are working.

Questions:
- Should we make this repo private? I didn't know that forked repo are automatically public and can only be public.

- [x] Updated Gradle version in `build.gradle`
- [x] Wrote new unit tests
- [x] Ran all unit tests to ensure they pass
- [x] Looked at the diff a final time before submitting to prevent common issues

### CODE REVIEW

File: src/main/java/de/mpii/clausie/ClauseDetector.java [hunk 1]
```diff
@@ -823,7 +823,7 @@ private static void addPartmodClause(ClausIE clausIE,
                                          IndexedWord verb,
                                          List<IndexedWord> roots) {
         IndexedWord partmodsub = subject;
-        addNsubjClause(clausIE, roots, clausIE.clauses, partmodsub, verb, false);
+        addNsubjClause(clausIE, roots, clausIE.clauses, partmodsub, verb, true);
```
Comment (AgoloKennethTan) [inline src/main/java/de/mpii/clausie/ClauseDetector.java]: In the original source code, this is false.

File: src/main/resources/clausie/clausie.conf [hunk 2]
```diff
@@ -6,14 +6,14 @@
 conservativeSVA = true
 conservativeSVOA = false
 processCcAllVerbs = true
-processCcNonVerbs = true
+processCcNonVerbs = false
```
Comment (AgoloKennethTan) [inline src/main/resources/clausie/clausie.conf:9]: according to the default used in Pyramid.

File: src/main/resources/clausie/clausie.conf [hunk 3]
```diff
@@ -6,14 +6,14 @@
...
 processAppositions = true
 appositionVerb = is
 processPossessives = true
 processPartmods = true
 possessiveVerb = has
 lemmatize = false
-nary = true
+nary = false
```
Comment (AgoloKennethTan) [inline src/main/resources/clausie/clausie.conf:16]: according to the default used in Pyramid.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,68 @@
+apply plugin: 'java'
```
Comment (AgoloKennethTan) [inline build.gradle:1]: Generated using `gradle init` and made some modification.

File: gradlew [hunk 5]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (AgoloKennethTan) [inline gradlew:1]: This file is auto-generated.

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#Tue May 01 16:18:15 EDT 2018
```
Comment (AgoloKennethTan) [inline gradle/wrapper/gradle-wrapper.properties:1]: This file is auto-generated.

File: gradlew.bat [hunk 7]
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (AgoloKennethTan) [inline gradlew.bat:1]: This file is auto-generated.

File: pom.xml [hunk 8]
```diff
@@ -1,141 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AgoloKennethTan) [inline pom.xml:1]: No longer needed. Since we are using gradle.

File: src/main/java/de/mpii/clausie/ClauseDetector.java [hunk 9]
```diff
@@ -316,7 +316,7 @@ private static void addNsubjClause(ClausIE clausIE,
                 IndexedWord dependent = outgoingEdge.getDependent();
 
                 // to avoid compl or mark in a main clause. "I doubt if she was sure whether this was important".
-                if (DpUtils.isComplm(outgoingEdge) || DpUtils.isMark(outgoingEdge)) {
+                if ((DpUtils.isComplm(outgoingEdge) || DpUtils.isMark(outgoingEdge)) && constRoot instanceof IndexedConstituent) {
```
Comment (AgoloKennethTan) [inline src/main/java/de/mpii/clausie/ClauseDetector.java:319]: Added check.

**Review by AgoloKennethTan — COMMENTED** [review]
@agoloprem, can you help me review this please?
Do let me know if you need me to go through the changes here. Maybe I can improve the documentation to denote the reverts and changes.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+- data/ : A set of test sentences
+- src/: Clausie source code
+
+## LICENSE
+- Clausie is distributed under the Attribution-ShareAlike (ver. 3.0 or later) [http://creativecommons.org/licenses/by-sa/3.0/legalcode]
+
+As explained in the license code, Clausie come with absolutely no warranty.
+
+# Changes Agolo made
```
Comment (agoloprem) [inline README.md:12]: Nice!

File: .gitignore [hunk 11]
```diff
@@ -52,4 +52,10 @@ ehthumbs.db
 [Tt]humbs.db
 /test-output
 
-hs_err_pid*
\ No newline at end of file
+hs_err_pid*
```
Comment (agoloprem) [inline .gitignore]: Did you get this from TemplateJavaProject? If not, please refer to the .gitignore there and copy/paste lines from that file.

File: .gitignore [hunk 12]
```diff
@@ -52,4 +52,10 @@ ehthumbs.db
...
+
+# Build
+.gradle
+gradle.properties
```
Comment (agoloprem) [inline .gitignore]: Good thinking ignoring this.

File: pom.xml (hunk 8 above)
Comment (agoloprem) [reply pom.xml:1]: 🎉

File: src/test/java/de/mpii/clausie/TestClausIE.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * @author Kenneth.Tan
+ * @since 2018-05-02
+ */
+@RunWith(JUnit4.class)
+public class TestClausIE {
```
Comment (agoloprem) [inline src/test/java/de/mpii/clausie/TestClausIE.java:17]: Good job, Kenneth!

**Review by agoloprem — APPROVED** [review]
I think it's OK to keep this public.

Ship it!

File: .gitignore (hunk 11 above)
Comment (AgoloKennethTan) [reply .gitignore]: Done! I also removed the ones that we don't need.

File: .gitignore (hunk 12 above)
Comment (AgoloKennethTan) [reply .gitignore]: Thanks :)

**Pull request merged**